datafusion/common/src/config.rs

```diff
@@ -22,12 +22,55 @@ use std::collections::{BTreeMap, HashMap};
 use std::error::Error;
 use std::fmt::{self, Display};
 use std::str::FromStr;
-
 use crate::error::_config_err;
 use crate::parsers::CompressionTypeVariant;
 use crate::utils::get_available_parallelism;
 use crate::{DataFusionError, Result};
 
+use serde::{Deserialize, Serialize};
+use hex;
+#[cfg(feature = "parquet")]
+use parquet::encryption::encrypt::FileEncryptionProperties;
+#[cfg(feature = "parquet")]
+use parquet::encryption::decrypt::FileDecryptionProperties;
+
+#[derive(Serialize, Deserialize)]
+pub struct EncryptionColumnKeys {
+    pub column_keys_as_hex: HashMap<String, String>,
+}
+
+impl EncryptionColumnKeys {
+    pub fn new(column_keys: &HashMap<String, Vec<u8>>) -> Self {
+        let mut column_keys_as_hex: HashMap<String, String> = HashMap::new();
+
+        for (key, value) in column_keys.iter() {
+            column_keys_as_hex.insert(hex::encode(key.clone()), hex::encode(value.clone()));
+        }
+
+        EncryptionColumnKeys {column_keys_as_hex}
+    }
+
+    pub fn to_json(&self) -> String {
+        serde_json::to_string(self).unwrap()
+    }
+
+    pub fn from_json_to_column_keys(s: &String) -> HashMap<String, Vec<u8>> {
+        let eck: EncryptionColumnKeys = serde_json::from_str(s).expect("failed to decode column keys from JSON");
+        let mut hm: HashMap<String, Vec<u8>> = HashMap::new();
+        for (key, val) in eck.column_keys_as_hex {
+            let col = String::from_utf8(hex::decode(key).expect("Invalid column name")).expect("Invalid column name");
+            let key = hex::decode(val).expect("Invalid column key");
+            hm.insert(col, key);
+        }
+        hm
+    }
+}
```

Comment:
Code to hold pure string version of column keys. Probably a bit overcomplex for what it is,